autopush/endpoint.py

```diff
@@ -220,3 +256,115 @@ def write_error(self, code, exception=None):
         if exception is not None:
             log.err(exception)
         self.finish()
+
+
+class RegistrationHandler(cyclone.web.RequestHandler):
+
+    def _error_response(self, failure):
+        log.err(failure)
+        self.set_status(500)
+        self.write("Error processing request")
+        self.finish()
+
+    def _load_params(self):
+        tags = {'chid': 'channelid',
+                'conn': 'connect',
+                }
+        chid = conn = None
+        if len(self.request.body) > 0:
```

Comment:
`EndpointHandler` uses this pattern, too...I wonder if it's worth breaking out into a generic "parse body with defaults" function.